active/0002-centalized-mgmt-resource-types-provider-versions.md

```diff
@@ -0,0 +1,258 @@
+
+## Detailed design
+
+### Client side changes
+
+#### Add a new syntax for the provider import declaration statements: `provider {providerName} [as {optionalAlias}]`
+
+The example below shows a usage of the proposed syntax
+
+```bicep
+provider az
+provider kubernetes as k8s // uses optional aliasing of the
+```
+
+The resolution of the symbols will be resolved from the builtin `bicepconfig.json` under a new section:
+
+```json
+{
+  // prior bicepconfig.json sections
+  // ...
+  "providers": {
+    // If you specify the entry `az` here you override the builtin `az` provider
+    "az": {
+      // A package reference string (below) gestures Bicep to source the types from the registry address.
+      "source": "mcr.microsoft.com/bicep/providers/az",
+      "version": "0.2.3"
+    },
+    "kubernetes": {
+      // Using the value "builtin" (below) gestures Bicep to source types from the statically compiled NuGet package reference
+      "builtIn": true
+    }
+  }
+  // ...
+}
+```
+
+Since legacy provider declaration syntax grammar continues to be supported, we introduce the following constraints:
+
+- To ensure consistency with pre-existing handling of `bicepconfig.json` the configuration file closest to the Bicep file in the directory hierarchy is used.
+- The `sys` identifier is coupled to the Bicep bits and its version cannot be overriden or dynamically uploaded using the mechanism described above
+  If a user specifies an entry called `sys` in the section above, this will be identified by the json schema as forbidden value. The reason is to prevent someone from overriding this reserved identifier.
+- The `az` identifier is implicitly imported into the global scope, the author can override the type definitions used in the namespace by adding an entry in the `providers` section (as shown above).
+- The keys of the `providers` object must all be unique. Notice how the source is the full repository path, so its possible to disambiguate in the case an author chosses to consume a provider with the same name from separate sources.
+- The keys of the `providers` section will be used as the identifiers in the `provider {providerName} [as {optionalAlias}]` statement
+
+The syntax below currently used to import built-in providers will be deprecated and result in a diagnostic as a consequence of the proposed changes in this proposal
+
+```bicep
+provider 'sys@1.0.0'
+provider 'kubernetes@1.0.0' as k8s
+```
+
+#### Change the delimiter '@' used by the supported provider declaration syntax to ':' to align with module reference syntax and increase consistency
+
+The syntax shown below uses ':' instead of '@' as a delimiter for provider declaration statements
+
+```bicep
+provider 'br:mcr.microsoft.com/bicep/providers/az:0.2.3' // notice its NOT using az@0.2.3
+provider 'br/public:az:0.2.3' as myAz
+```
+
+## Drawbacks
+
+- Costumers may be confused by the configt merge semantics behavior
```

Comment:
[nit] Costumers -> Customers here and in multiple other places. I also generally prefer to just say "users" rather than "customers", as I feel like it's more appropriate for an open-source project (less "us" vs "them")